examples/viewer_fps_label.py

```diff
@@ -19,7 +19,9 @@ def update_fps(fps):
 viewer = napari.Viewer()
 viewer.add_image(np.random.random((5, 5, 5)), colormap='red', opacity=0.8)
 viewer.text_overlay.visible = True
-viewer.window.qt_viewer.canvas.measure_fps(callback=update_fps)
+# note: this is using a private attribute, so it might break
+# without warningin future versions!
+viewer.window._qt_viewer.canvas._scene_canvas.measure_fps(callback=update_fps)
```

Comment:
Why we cannot add a backward compatibility layer for this?

Reply:
But pure QtViewer is part of public API (used directly, for example in PartSeg). And preferred way to embed napari in another application.

Reply:
Uhm... Then I guess we need a deprecation for `qtviewer.canvas` itself, which is a lot more complicated given the amount of changes we have here. Not sure how to go about this in terms of deprecation, cause `QtViewer.canvas` is being replaced by a completely different object. Maybe we can temporarly call this new one `_canvas` and have `canvas` emit a `DeprecationWarning` when accessed that?

Reply:
That was also my impression indeed, but I do see that in the public qt __init__ __all__ also states the QtViewer.

In any case, please correct me if I am wrong, but would backward compatibility not be achieved by creating a property to getting canvas.camera and canvas.view within QtViewer? The deprecation warning could then be added here. The interaction with canvas.bgcolor, canvas.background_color_override etc has not been changed and should be backward compatible already.

@Czaki @brisvag It would be nice to hear your thoughts about this.

Reply:
See [2630945](https://github.com/napari/napari/pull/5432/commits/26309455559a77c3a6133539350eee759692d9fe) for backward compatibility. I believe this would allow for enough backward compatibility. 

The warnings specifically say ```napari.qt``` to indicate the public Qt used in plugins.